watertap/core/zero_order_properties.py

```diff
@@ -0,0 +1,304 @@
+                    hold_state=True.
+            outlvl : sets output level of of logging
+        '''
+        init_log = idaeslog.getInitLogger(blk.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+
+        # Unfix state variables
+        revert_state_vars(blk, flags)
+        init_log.info('State Released.')
+
+
+@declare_process_block_class("WaterStateBlock",
+                             block_class=_WaterStateBlock)
+class WaterStateBlockData(StateBlockData):
+    """
+    General purpose StateBlock for Zero-Order unit models.
+    """
+
+    def build(self):
+        super().build()
+
+        # Create state variables
+        self.flow_vol = Var(initialize=1.0,
+                            domain=PositiveReals,
+                            doc='Total volumetric flowrate',
+                            units=pyunits.m**3/pyunits.s)
+        self.pressure = Var(domain=PositiveReals,
+                            initialize=101325.0,
+                            bounds=(1e3, 1e6),
+                            doc='Pressure',
+                            units=pyunits.Pa)
+        self.temperature = Var(domain=PositiveReals,
+                               initialize=298.15,
+                               bounds=(298.15, 323.15),
+                               doc='Temperature',
+                               units=pyunits.K)
+        self.conc_mass_comp = Var(self.params.solute_set,
+                                  domain=PositiveReals,
+                                  initialize=1e-5,
+                                  bounds=(1e-20, 1e3),
+                                  doc='Component mass concentrations',
+                                  units=pyunits.kg/pyunits.m**3)
+
+        # ---------------------------------------------------------------------
+        # Flow and density expressions
+        def rule_flow_mass_comp(blk, j):
+            if j == "H2O":
+                return blk.flow_vol*blk.params.dens_mass
+            else:
+                return blk.flow_vol*blk.conc_mass_comp[j]
+        self.flow_mass_comp = Expression(
+            self.component_list, rule=rule_flow_mass_comp)
+
+        def rule_enth_dens(blk):
+            return (blk.params.dens_mass*blk.params.cp_mass *
+                    (blk.temperature - blk.params.temperature_ref))
+        self._enth_dens_term = Expression(
+            rule=rule_enth_dens)
+
+        def rule_enth_flow(blk):
+            return blk.flow_vol*blk._enth_dens_term
+        self._enth_flow_term = Expression(
+            rule=rule_enth_flow)
+
+    # -------------------------------------------------------------------------
+    # Other properties
+    def _cp_mass(self):
+        add_object_reference(self, "cp_mass", self.params.cp_mass)
+
+    def _dens_mass(self):
+        add_object_reference(self, "dens_mass", self.params.dens_mass)
+
+    def get_material_flow_terms(blk, p, j):
```

Comment:
Do we need the `p` here?

Reply:
Yes, we do. It is a part of the framework API, so when this method is called it will always be passed `p` (even though we don't actually need it in this case).